extensions/default/src/Toolbar/ToolbarLayoutSelector.tsx

```diff
@@ -61,12 +61,31 @@ function LayoutSelector({ rows, columns, servicesManager }) {
       label="Grid Layout"
       icon="tool-layout"
       onInteraction={onInteractionHandler}
+      className={className}
+      rounded={rest.rounded}
       dropdownContent={
         DropdownContent !== null && (
           <DropdownContent
             rows={rows}
             columns={columns}
             onSelection={({ numRows, numCols }) => {
+              if (
+                HangingProtocolService.getActiveProtocol().protocol.id === 'mpr'
+              ) {
+                ToolBarService.recordInteraction({
+                  groupId: 'MPR',
+                  itemId: 'MPR',
+                  interactionType: 'toggle',
+                  commands: [
+                    {
+                      commandName: 'toggleMPR',
+                      commandOptions: {},
+                      context: 'CORNERSTONE',
+                    },
+                  ],
+                });
+              }
+
               viewportGridService.setLayout({ numRows, numCols });
```

Comment:
I think the right way to do this is to call recordInteraction on the grid layout button, and that should consist of two items, first a toggleMPR item, and second setLayout command.  That way, modes can configure additional custom behaviour if they want, and it avoids needing to know about any special behaviour here.  See more comments below.

Reply:
See previous comment [regarding StateSyncService.](https://github.com/OHIF/Viewers/pull/3086#discussion_r1062566151)